src/main/java/rx/internal/operators/OnSubscribeOnAssembly.java

```diff
@@ -40,15 +41,14 @@
     
     public OnSubscribeOnAssembly(OnSubscribe<T> source) {
         this.source = source;
-        this.stacktrace = createStacktrace();
+        this.assemblyStacktrace = assemblyStacktrace();
     }
     
-    static String createStacktrace() {
-        StackTraceElement[] stes = Thread.currentThread().getStackTrace();
+    static StackTraceElement[] assemblyStacktrace() {
+        final StackTraceElement[] in = new Exception().getStackTrace();
+        final List<StackTraceElement> out = new ArrayList<StackTraceElement>(in.length);
 
-        StringBuilder sb = new StringBuilder("Assembly trace:");
-        
-        for (StackTraceElement e : stes) {
+        for (StackTraceElement e : in) {
```

Comment:
I'll open separate PR after this one where I'll move filtering stacktrace to `onError`. This is a hot path while `onError` isn't, so we can save some CPU time on filtering for errorless chains.